services/autoscaling/src/simcore_service_autoscaling/models.py

```diff
@@ -84,3 +85,33 @@ class EC2InstanceData:
 class AssociatedInstance:
     node: Node
     ec2_instance: EC2InstanceData
+
+
+@dataclass(frozen=True)
+class Cluster:
+    active_nodes: list[AssociatedInstance] = field(
+        metadata={
+            "description": "This is a EC2 backed docker node which is active (with running tasks)"
+        }
+    )
+    drained_nodes: list[AssociatedInstance] = field(
+        metadata={
+            "description": "This is a EC2 backed docker node which is drained (with no tasks)"
+        }
+    )
+    reserve_drained_nodes: list[AssociatedInstance] = field(
+        metadata={
+            "description": "This is a EC2 backed docker node which is drained in the reserve if this is enabled (with no tasks)"
+        }
+    )
+    pending_ec2s: list[EC2InstanceData] = field(
+        metadata={
+            "description": "This is an EC2 instance that is not yet associated to a docker node"
+        }
+    )
+    disconnected_nodes: list[Node] = field(
+        metadata={
+            "description": "This is a docker node which is not backed by a running EC2 instance"
+        }
+    )
+    terminated_instances: list[EC2InstanceData]
```

Comment:
MINOR metadata.description here?
Q: Is this dataclass field metadata used somewhere in the code (e.g. to generate specs/reports... or just for doc ?

Reply:
no that is currently documentation only. I like having this in the pydantic models, now I can do the same in the dataclasses when these are complex objects.

Reply:
THOUGHT: i always wondered what is the impact of these in memory vs a simple comment provided that they are used for doc.